lib/pbench/server/api/resources/upload_api.py

```diff
@@ -398,18 +403,6 @@ def _put(self, args: ApiParams, request: Request, context: ApiContext) -> Respon
             # NOTE: we're setting the Dataset "created" timestamp here, but it
             # won't be committed to the database until the "advance" operation
             # at the end.
-            try:
-                metadata = tarball.get_metadata()
-                dataset.created = UtcTimeHelper.from_string(
-                    metadata["pbench"]["date"]
-                ).utc_time
-                Metadata.create(dataset=dataset, key=Metadata.METALOG, value=metadata)
-            except Exception as exc:
-                raise CleanupTime(
-                    HTTPStatus.BAD_REQUEST,
-                    f"Tarball {dataset.name!r} is invalid or missing required metadata.log: {exc}",
-                )
-
```

Comment:
The alternative, which wouldn't really simplify anything but would possibly make the cache manager interface look a tiny bit prettier (I don't believe it's reasonable to care yet), would be to rely on the cache manager `create` itself to unpack `metadata.log` from the tarball. I'd considered that approach, and decided that wasn't a good idea for this pass since `PUT` is already unpacking it.

We are very definitely not ready to remove `controller` from lower levels: it's the basis of the file system structure, and changing this will break the indexer, which we do not want to address in this PR.

Reply:
I'm very much in favor of designing and implementing the _interfaces_ that we want to use, even if they don't immediately yield the best implementation.

If `CacheManager.create()` needs to come up with a "real" value for the controller (at least, in the interim), that's OK with me (although, I suspect that using a constant value would suffice), but I don't think it should be provided as a parameter (especially, since the caller then has to go through the same motions that `create()` would have in order to produce it).

That is, if we're going to implement throw-away code, lets put it exactly where it is needed and not in that function's caller.  (Yes, it might mean duplicating the access to the metadata (or it might not...), but when the controller is no longer needed, we won't have to disturb the calling code when we rip it out.)

Reply:
Riya ran into a chicken-and-egg issue, and addressed it cleanly with minimal interface changes. I like that, and in general would like to encourage that rather than taking a superficial change like dropping an HTTP header as an excuse to radically refactor a loosely related class and possibly rototill the historical file system hierarchy.

We'll be doing all that: this isn't the time. The cache manager interfaces will *definitely* be changing, substantially, in order to fulfill its mission as we move into dynamic tracked unpacking. Making an out-of-scope change now won't reduce the churn later: it just increases the churn now.

Reply:
Understood, and I don't disagree...which is why I suggested removing the controller value entirely from `_put()`, rather than taking the extra step to produce it so that we don't have to change the call to `CacheManager.create()`:  given that with this change `_put()` will no longer have direct access to the controller value (i.e., from the HTTP headers), it makes as much sense as anything to relegate the problem/change of producing it to the `CacheManager.create()` code as it does to burden the `_put()` code with it.  (In particular, the result is _less_ change to the `_put()` code, which is the stuff that we're going to "keep".)  And, then `create()` can produce it any way it wants (e.g., it can probably just use a constant rather than trying to come up with the real value).